app/assets/javascripts/provider/admin/apiconfig/services/service_discovery.js.coffee

```diff
@@ -0,0 +1,59 @@
+$ ->
+  form_source = $('form#new_service_source')[0]
+  form_scratch = $('form#new_service,form#create_service')[0]
+  form_discover = $('form#service_discovery')[0]
+
+  clear_select_options = (select)->
+    $(select).find('option').not(':first').remove()
+
+  clear_namespaces = ()->
+    clear_select_options form_discover.service_namespace
```

Comment:
Hmm so you create an attribute service_namespace?
Why not using `data` attribute?